src/main/plots/solution_graph_statistics_plots.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2020 Anastasiia Birillo, Elena Lyulina
+
+from itertools import groupby
+from typing import Dict, List
+from collections import defaultdict
+
+import plotly.graph_objects as go
+
+import pandas as pd
+import plotly.express as px
+
+from src.main.util import consts
+from src.main.canonicalization.consts import TREE_TYPE
+from src.main.solution_space.solution_graph import SolutionGraph
+from src.main.plots.util.plotly_util import save_plot, get_freq_bar_chart, update_layout
+from src.main.plots.util.plots_common import get_readable_key, get_labels_for_freq_plots
+from src.main.statistics_gathering.solution_graph_statistics import get_general_solution_graph_statistics, \
+    get_statistics_for_each_vertex
+from src.main.plots.util.consts import STATISTICS_KEY, PLOTTY_CATEGORY_ORDER, STATISTICS_FREQ, STATISTICS_SHOWING_KEY, \
+    STATISTICS_COLORS, PLOT_TYPE, DEFAULT_PATH_FOR_PLOTS
+
+
+def __plot_general_statistics(statistics_dict: Dict[int, int], title: str, path: str, labels: Dict[str, str],
+                              plot_name: str = 'general_statistics_plot',
+                              column: STATISTICS_KEY = STATISTICS_KEY.NODES_NUMBER,
+                              format: consts.EXTENSION = consts.EXTENSION.HTML, auto_open: bool = False,
+                              x_category_order: PLOTTY_CATEGORY_ORDER = PLOTTY_CATEGORY_ORDER.CATEGORY_ASCENDING) -> None:
+    statistics_df = pd.DataFrame(statistics_dict.items(), columns=[column.value, STATISTICS_FREQ])
+    # x_category_order='total ascending' means: in order of increasing values in Y
+    # x_category_order='category ascending' means: in order of increasing values in X
+    fig = px.bar(statistics_df, x=column.value, y=STATISTICS_FREQ, title=title, labels=labels,
+                 hover_data=[column.value, STATISTICS_FREQ])
+    fig.update_layout(
+        yaxis=dict(
+            title_text=STATISTICS_SHOWING_KEY.FREQ.value
+        ),
+        xaxis=dict(
+            title_text=get_readable_key(column.value),
+            categoryorder=x_category_order.value
+        ),
+        plot_bgcolor=STATISTICS_COLORS.BAR_CHART_BG.value
+    )
+    fig.update_yaxes(automargin=True)
+    save_plot(fig, path, PLOT_TYPE.BAR, plot_name, format, auto_open)
+
+
+def plot_general_statistics(solution_graph: SolutionGraph, path: str = DEFAULT_PATH_FOR_PLOTS,
```

Comment:
'General statistics' is a little bit unclear because it can mean anything. How about 'plot_all_node_numbers_freq'?